src/Corporation/Actions.ts

```diff
@@ -396,14 +396,8 @@ export function MakeProduct(
   if (corp.funds < designInvest + marketingInvest) {
     throw new Error("You don't have enough company funds to make this large of an investment");
   }
-  let maxProducts = 3;
-  if (division.hasResearch("uPgrade: Capacity.II")) {
-    maxProducts = 5;
-  } else if (division.hasResearch("uPgrade: Capacity.I")) {
-    maxProducts = 4;
-  }
-  if (division.products.size >= maxProducts) {
-    throw new Error(`You are already at the max products (${maxProducts}) for division: ${division.name}!`);
+  if (division.products.size >= division.maxProducts) {
+    throw new Error(`You are already at the max products (${division.maxProducts}) for division: ${division.name}!`);
```

Comment:
Not a big deal here as this is an error path and doesn't really need to be optimized, but this is calling the same function twice in a row to calculate max products (through multiple references to the getter variable).

I also like how nice getters can look, but in general they make it harder to notice inefficiencies like this, where if it was regular function call syntax, it would be clear here that this code path is calling getMaxProducts() or whatever twice in a row, which might normally prompt us to make a variable so we can reuse the value. Just mentioning it as this is a reason generally why we shouldn't broadly expand use of getters, but no need to revert the use here.